modules/internal/localeModels/knl/LocaleModel.chpl

```diff
@@ -287,6 +285,9 @@ module LocaleModel {
       const origSubloc = chpl_task_getRequestedSubloc();
       chpl_task_setSubloc(hbm_id);
       hbm = new MemoryLocale(hbm_id, this);
+
+      super.init(_parent);
+
       chpl_task_setSubloc(origSubloc);
```

Comment:
I believe the super.init() should be moved down one line so we will be in the original sublocale when it is called.  Note that we only change the sublocale for one brief moment when we create the MemoryLocale for the high-bandwidth memory, and then we change it back.

I missed this in the original PR because I was focused on understanding what the super.init() did.  Sorry about that.